### PR TITLE
Make LearningModule <-> Task a HABTM relationship

### PR DESCRIPTION
**Pull request by joshlory**
Probably easier to review this PR commit-by-commit.  The editor for the HABTM association is a multi-select:

habtm_learning_modules_tasks

The migration is reversible.  Rails only creates the new HABTM table, so the old associations are migrated manually via SQL.

```
mysql> select id, plc_learning_module_id from plc_tasks;
+----+------------------------+
| id | plc_learning_module_id |
+----+------------------------+
|  1 |                      1 |
|  2 |                      1 |
|  3 |                      1 |
|  4 |                      1 |
|  5 |                      1 |
|  6 |                      2 |
+----+------------------------+
6 rows in set (0.00 sec)

$ rake db:migrate

mysql> select * from plc_learning_modules_tasks;
+------------------------+-------------+
| plc_learning_module_id | plc_task_id |
+------------------------+-------------+
|                      1 |           1 |
|                      1 |           2 |
|                      1 |           3 |
|                      1 |           4 |
|                      1 |           5 |
|                      2 |           6 |
+------------------------+-------------+
6 rows in set (0.00 sec)

$ rake db:rollback STEP=1

(Same as initial output)
```

### CODE REVIEW

**Comment by mehalshah** [comment]
So relationship is now many to many, and we get to tasks by going through any of the modules they are associated with, right?

There may be a justification for an index view for tasks at this point

File: dashboard/app/controllers/plc/tasks_controller.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ def show
```
Comment (mehalshah) [inline dashboard/app/controllers/plc/tasks_controller.rb:8]: The entry point for task creation is still from the learning module page. You might as well start with the learning_module_id preselected

File: dashboard/app/views/plc/tasks/new.html.haml [hunk 2]
```diff
@@ -2,7 +2,4 @@
 
 = render 'form'
```
Comment (mehalshah) [inline dashboard/app/views/plc/tasks/new.html.haml:4]: If you got to this page with a learning_module_id param, might as well let the content creator go back to that module

File: dashboard/app/views/plc/tasks/edit/_common_editable.html.haml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 = f.label :name
 = f.text_field :name
-= f.label :learning_module_id
-= f.select :plc_learning_module_id, options_for_select(options_for_plc_task_learning_modules, @task.plc_learning_module_id)
+= f.label :learning_modules
+= f.select :plc_learning_module_ids, options_for_select(options_for_plc_task_learning_modules, @task.plc_learning_modules.pluck(:id)), {}, {multiple: true, size: 5}
```
Comment (mehalshah) [inline dashboard/app/views/plc/tasks/edit/_common_editable.html.haml]: Make this larger and wider. It won't accommodate modules with long names. Also 5 is a small number and makes for a lot of scrolling. Maybe make size min(25, # modules) or something?

Alternatively, use selectize. But we don't _need_ that

File: dashboard/app/views/plc/tasks/show.html.haml [hunk 4]
```diff
@@ -6,13 +6,13 @@
   = @task.name
 
 %p
-  %strong Associated module
-  = @task.plc_learning_module.try(:name)
+  %strong Associated modules:
+%ul
+  - @task.plc_learning_modules.pluck(:id, :name).each do |id, name|
```
Comment (mehalshah) [inline dashboard/app/views/plc/tasks/show.html.haml]: Alphabetically sort this. (We should start alphabetically sorting elements that are peers of each other)

File: dashboard/app/controllers/plc/tasks_controller.rb (hunk 1 above)
Comment (joshlory) [inline dashboard/app/controllers/plc/tasks_controller.rb:8]: I'm surprised it wasn't preselected.  It looks like `.pluck` doesn't work if the `@task` hasn't been saved yet.  I'll update to use `.map` for the `TaskController#new`.

**Comment by joshlory** [comment]
@laurelfan can you take a look at this migration?  Thanks!

File: dashboard/app/controllers/plc/tasks_controller.rb (hunk 1 above)
Comment (joshlory) [inline dashboard/app/controllers/plc/tasks_controller.rb:8]: Fixed in 34d208f765804b6d99cc5323c693ea7615188392.

File: dashboard/app/views/plc/tasks/edit/_common_editable.html.haml (hunk 3 above)
Comment (joshlory) [inline dashboard/app/views/plc/tasks/edit/_common_editable.html.haml]: Fixed in 34d208f765804b6d99cc5323c693ea7615188392.

File: dashboard/app/views/plc/tasks/show.html.haml (hunk 4 above)
Comment (joshlory) [inline dashboard/app/views/plc/tasks/show.html.haml]: Fixed in 34d208f765804b6d99cc5323c693ea7615188392.

File: dashboard/app/views/plc/tasks/new.html.haml (hunk 2 above)
Comment (joshlory) [inline dashboard/app/views/plc/tasks/new.html.haml:4]: Fixed in 34d208f765804b6d99cc5323c693ea7615188392.

File: dashboard/db/migrate/20160406190753_create_join_table_plc_learning_modules_plc_tasks.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    execute 'insert into plc_learning_modules_tasks (plc_learning_module_id, plc_task_id) select plc_learning_module_id, id from plc_tasks;'
+    remove_column :plc_tasks, :plc_learning_module_id
+  end
+
+  def down
+    add_column :plc_tasks, :plc_learning_module_id, :integer
+    execute 'update plc_tasks t inner join plc_learning_modules_tasks j on t.id = j.plc_task_id set t.plc_learning_module_id = j.plc_learning_module_id;'
+    drop_table :plc_learning_modules_tasks
+  end
+end
```
Comment (joshlory) [inline dashboard/db/migrate/20160406190753_create_join_table_plc_learning_modules_plc_tasks.rb:16]: @laurelfan can you take a look at this migration?  Thanks!

File: dashboard/db/migrate/20160406190753_create_join_table_plc_learning_modules_plc_tasks.rb [hunk 6]
```diff
@@ -0,0 +1,16 @@
+class CreateJoinTablePlcLearningModulesPlcTasks < ActiveRecord::Migration
+  def up
+    create_join_table :plc_learning_modules, :plc_tasks do |t|
+      t.index :plc_learning_module_id
+      t.index :plc_task_id
+    end
+    execute 'insert into plc_learning_modules_tasks (plc_learning_module_id, plc_task_id) select plc_learning_module_id, id from plc_tasks;'
```
Comment (mehalshah) [inline dashboard/db/migrate/20160406190753_create_join_table_plc_learning_modules_plc_tasks.rb:7]: You can combine lines 3 to 7 with one "create table as " statement if desired. Not sure if it actually is desired

**Comment by mehalshah** [comment]
Looks good. Optionally - you can simplify Plc::CourseUnit.get_all_possible_learning_resources while you are doing this. But it's no big deal

**Comment by joshlory** [comment]
I'll open a new PR for cleanup based on this new association (also once `CourseUnit -> LearningModule` is done).

**Pull request merged**